### PR TITLE
[BAEL-3319] - Documenting a Spring REST API using OpenAPI

### PR DESCRIPTION
**Pull request by harshavs**

### CODE REVIEW

File: spring-boot-springdoc/src/main/java/com/baeldung/springdoc/controller/BookController.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    private BookRepository repository;
+
+    @GetMapping("/{id}")
+    public Book findById(@PathVariable long id) {
+        return repository.findById(id)
+            .orElseThrow(() -> new BookNotFoundException());
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/")
+    @ResponseBody
```
Comment (KevinGilmore) [inline spring-boot-springdoc/src/main/java/com/baeldung/springdoc/controller/BookController.java]: Also when using RestController annotation, we don't need ResponseBody annotation.

File: spring-boot-springdoc/src/main/java/com/baeldung/springdoc/repository/BookRepository.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import java.util.Optional;
+
+import org.springframework.stereotype.Repository;
+
+import com.baeldung.springdoc.model.Book;
+
+@Repository
+public class BookRepository {
+
+    private List<Book> books = new ArrayList<>();
```
Comment (KevinGilmore) [inline spring-boot-springdoc/src/main/java/com/baeldung/springdoc/repository/BookRepository.java]: Obviously this is not the main point of the article, but I would probably use a Map instead of a List here.

File: spring-webflux-amqp/pom.xml [hunk 3]
```diff
@@ -62,6 +62,12 @@
 			<groupId>org.springframework.boot</groupId>
 			<artifactId>spring-boot-starter-integration</artifactId>
 		</dependency>
+
```
Comment (KevinGilmore) [inline spring-webflux-amqp/pom.xml]: And yes, we should revert this change.

File: spring-boot-springdoc/pom.xml [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+	<modelVersion>4.0.0</modelVersion>
+	<groupId>com.baeldung</groupId>
+	<artifactId>spring-boot-springdoc</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
+	<name>spring-boot-springdoc</name>
+	<description>Project for Springdoc integration</description>
+	<packaging>war</packaging>
```
[the remark this answers is not in the document]
Comment (harshavs) [reply spring-boot-springdoc/pom.xml]: It was jar originally. I forgot to change it back after my testing.

File: spring-boot-springdoc/src/main/java/com/baeldung/springdoc/controller/BookController.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        return book;
+    }
+
+    @RequestMapping(method = RequestMethod.HEAD, value = "/")
+    @ResponseStatus(HttpStatus.OK)
+    public Book headBook() {
+        return new Book();
+    }
+
+    @RequestMapping(method = RequestMethod.DELETE, value = "/{id}")
```
Comment (KevinGilmore) [inline spring-boot-springdoc/src/main/java/com/baeldung/springdoc/controller/BookController.java]: You should be able to use DeleteMapping here.

File: spring-boot-springdoc/pom.xml [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+        <relativePath>../parent-boot-2</relativePath>
+    </parent>
+
+    <properties>
+        <java.version>1.8</java.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
```
Comment (KevinGilmore) [inline spring-boot-springdoc/pom.xml]: If you're already using a specific starter, like spring-boot-starter-web, I don't think you have to explicitly include just plain spring-boot-starter.
